compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -1443,27 +1443,27 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
       /* organize constructors in a called-by tree
        * (the implicit root is the primary constructor)
        */
-      val secondaryCtorTrees = {
+      val ctorTree = {
         val ctorToChildren = secondaryCtors
           .groupBy(_.targetCtor)
           .withDefaultValue(Nil)
 
-        /* when constructing the call-by tree, we use in-order traversal to
+        /* when constructing the call-by tree, we use post-order traversal to
```

Comment:
I think the new walk is a pre-order, isn't it? First `overloadNum` is computed and `nextOverloadNum` is updated, before recursing into children.